### PR TITLE
docs: add Data retention page

### LINKED ISSUE #243: Document retention policy
In "operation" section of the doc we need to explain how to configure and maintain data retention. We have `alter table drop partition` SQL for that.

The documentation piece need to go thru the problem (retention, disk space saving etc) and how QuestDB can help there. This is separate from syntax page

### PR DESCRIPTION
**Pull request by bsmth**
__Additions:__
* data retention with examples for generating test data and dropping by date

~~__CSS Change:__~~
~~* Proposed reduction of vertical whitespace to improve readability~~

closes #243

### CODE REVIEW

**Comment by bsmth** [comment]
@mpsq there are some style changes here, feedback welcome

File: docs/operations/data-retention.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
+---
```
Comment (mpsq) [inline docs/operations/data-retention.md:1]: Could you also add that page to the list on "introduction"?

File: docs/operations/data-retention.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+This page provides a high-level overview of partitioning with examples to drop
+data by date. For more details on partitioning, see the
+[partitioning](/docs/concept/partitions/) page.
+
+## Strategy for data retention
+
+A simple approach to removing stale data is to drop data that has been
+partitioned by time. A table must have a
+[designated timestamp](/docs/concept/designated-timestamp) assigned and a
```
Comment (mpsq) [inline docs/operations/data-retention.md:22]:
```suggestion
[designated timestamp](/docs/concept/designated-timestamp/) assigned and a
```

File: docs/operations/data-retention.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+:::caution
+
+Use `DROP PARTITION` with care as QuestDB **cannot recover data from dropped
+partitions**!
+
+:::
+
+To drop partitions, users can use the
+[ALTER TABLE DROP PARTITION](/docs/reference/sql/alter-table-drop-partition)
```
Comment (mpsq) [inline docs/operations/data-retention.md:52]:
```suggestion
[ALTER TABLE DROP PARTITION](/docs/reference/sql/alter-table-drop-partition/)
```

**Review by mpsq — CHANGES_REQUESTED** [review]
minor tweaks

**Comment by bsmth** [comment]
> @mpsq there are some style changes here, feedback welcome

reverted in 62b17e9

File: docs/operations/data-retention.md (hunk 1 above)
Comment (bsmth) [reply docs/operations/data-retention.md:1]: done

**Review by mpsq — DISMISSED** [review]
LGTM

**Review by mpsq — APPROVED** [review]

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**